backend/src/Tabulators/StarResults.js

```diff
@@ -1,759 +1,253 @@
-const minScore = 0;
-const maxScore = 5;
-
-function parseData(header, data, nWinners) {
-  // Inspect the data to determine the type of data in each column
-  const transforms = getTransforms(header, data);
-
-  // The list of candidates is based on the columns containing STAR scores
-  const candidates = [];
-  for (let i = 0; i < transforms.length; i++) {
-    if (transforms[i] === transformScore) {
-      const candidate = {
-        name: transformAny(header[i]),
-        index: candidates.length,
-        csvColumn: i,
-        totalScore: 0,
-        support: new Array(6).fill(0)
-      };
-      candidates.push(candidate);
+const ParseData = require("./ParseData");
```

Comment:
This is probably the most important file of the project :)

Would lean to convert the whole thing to Typescript and add more comments. Also random q: can we have it set up so that different precincts can crunch their data separately, and then combine to see the final result?

BUT: totally legit to add this to our TODO list

Reply:
I didn't initially convert to Typescript because I wanted to eventually push this back to the STAR Core repo and have it be its own javascript npm package. But after learning a bit more it should be easy to compile it back to javascript and offer both, right? 

On the precinct question, yes we can! The python implementation currently lets you optionally pass in the summaryData instead of the ballots. So we could run the election for each precinct, sum up the data, then pass that into the tabulation function to get the total results. That way you'd get details for each precinct and the final results. It wouldn't be too much work, but lets wait until we add more precinct support in other areas of the project first.